src/utils.jl

```diff
@@ -370,42 +377,44 @@ function _transfer_solution!(data,pv,fh,ip_geo,ip_field,cellset_,val_buffer,val,
     grid = dh.grid
     # actual data
     local_dof_range = Ferrite.dof_range(fh, field_name)
-    _processreturn = length(process(val_buffer))
-
-    current_vertex_index = 1
+    _processreturndim = length(process(val_buffer))
 
     cell_geo_ref = Ferrite.getcells(grid, cellset_[1])
 
-    Ferrite.reinit!(pv, Ferrite.getcoordinates(grid,cellset_[1]), Tensors.Vec{dim}(ref_coords[current_vertex_index,:]))
+    Ferrite.reinit!(pv, Ferrite.getcoordinates(grid,cellset_[1]), Tensors.Vec{dim}(ref_coords[1,:]))
     n_basefuncs = Ferrite.getnbasefunctions(pv)
 
     _local_coords = Ferrite.getcoordinates(grid,cellset_[1])
     _local_celldofs = Ferrite.celldofs(dh,cellset_[1])
     _celldofs_field = reshape(@view(_local_celldofs[local_dof_range]), (field_dim, n_basefuncs))
     _local_ref_coords = Tensors.Vec{dim}(ref_coords[1,:])
 
+    # We just loop over all cells
     for (isvisible,(cell_idx,cell_geo)) in zip(plotter.visible,enumerate(Ferrite.getcells(dh.grid)))
-        # This should make the loop work for mixed grids
+        # Skip invisible cells and cells which are not in the current cellset
         if !isvisible || cell_idx ∉ cellset_
-            current_vertex_index += ntriangles(cell_geo)*n_vertices_per_tri
             continue
         end
+
+        # Buffer cell data relevant for the current field to transfer
         Ferrite.getcoordinates!(_local_coords,grid,cell_idx)
         Ferrite.celldofs!(_local_celldofs,dh,cell_idx)
         _celldofs_field = reshape(@view(_local_celldofs[local_dof_range]), (field_dim, n_basefuncs))
-        ncvertices = ntriangles(cell_geo)*n_vertices_per_tri
-        # TODO replace this with a triangle-to-cell map.
-        for i in 1:ncvertices
-            _local_ref_coords = Tensors.Vec{dim}(@view(ref_coords[current_vertex_index,:]))
-            Ferrite.reinit!(pv, _local_coords, _local_ref_coords)
-            for d in 1:field_dim
-                val_buffer[d] = Ferrite.function_value(pv, 1, @views(u[_celldofs_field[d,:]]))
-            end
-            val = process(val_buffer)
-            for d in 1:_processreturn
-                data[current_vertex_index,d] = val[d]
+
+        # Loop over the triangles of the cell and interpolate at the vertices
+        # TODO remove redundant function value calls
+        for triangle_index in triangles_on_cell(plotter, cell_idx)
+            for current_vertex_index in plotter.all_triangles[triangle_index]
```

Comment:
`all_triangles` renders also the ones that aren't visible right? So isn't here the skip missing if its visible or not?

Reply:
Huh? I have not changed the logic here and thought that `vis_triangles` is the makie buffer for rendering where invisible triangles get the vertices 1,1,1?

Reply:
ah right its above, nvm